### PR TITLE
PIN-3491 Catalog Process Migration

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala [hunk 1]
```diff
@@ -96,6 +98,26 @@ object ReadModelQueries {
...
+  )(implicit ec: ExecutionContext): Future[Option[CatalogDocument]] = {
+    val filters = Filters.and(
+      Filters.eq("data.id", eServiceId.toString),
+      Filters.eq("data.descriptors.id", descriptorId.toString),
+      Filters.or(
+        Filters.eq("data.descriptors.docs.id", documentId.toString),
+        Filters.eq("data.descriptors.interface.id", documentId.toString)
+      )
+    )
+    readModel.findOne[CatalogDocument]("eservices", filters)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala]: uhm are you sure this is working? the query should return a `CatalogItem` but it is trying to decode it as `CatalogDocument`

File: src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala [hunk 2]
```diff
@@ -1,69 +1,27 @@
...
-    controller.foreach(startServer)
-  }
-
-  override def afterAll(): Unit = {
-    shutDownServer()
-    super.afterAll()
-  }
-
-  "flatten e-service filter" must {
+  "flatten e-service filter" should {
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala]: this test seems quite...useless 🤣 
you can remove it, thanks!

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 3]
```diff
@@ -275,9 +280,15 @@ final case class ProcessApiServiceImpl(
...
 
-    val result: Future[EServiceDoc] = catalogManagementService
-      .getEServiceDocument(eServiceId, descriptorId, documentId)
-      .map(Converter.convertToApiEserviceDoc)
+    val result: Future[EServiceDoc] = for {
+      eServiceUuid    <- eServiceId.toFutureUUID
+      descriptorUuid  <- descriptorId.toFutureUUID
+      documentIdUuid  <- documentId.toFutureUUID
+      catalogItem     <- ReadModelQueries.getEService(eServiceUuid)(readModel)
+      _               <- catalogItem.toFuture(EServiceNotFound(eServiceId))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: minor enhancement, you can probably just retrieve the document, returning 404 if no results are returned
```suggestion
```

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 4]
```diff
@@ -457,13 +508,14 @@ final case class ProcessApiServiceImpl(
 
     val result: Future[EService] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
-      eService       <- catalogManagementService.getEService(eServiceId)
+      eServiceUuid   <- eServiceId.toFutureUUID
+      descriptorUuid <- descriptorId.toFutureUUID
+      catalogItem    <- ReadModelQueries
+        .getEService(eServiceUuid)(readModel)
+      eService       <- catalogItem.toFuture(EServiceNotFound(eServiceId))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: these two expressions are used very often, we could add a little helper method for this

**Review by galales — COMMENTED** [review]
LGTM overall, just a couple of comments 👍

File: src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala (hunk 1 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala:112]: I need to implement a aggregateRaw sorry.

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala (hunk 3 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala:288]: You are....right...

File: src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala (hunk 1 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala:112]: Better to do a filtering on CatalogItem

**Review by galales — APPROVED** [review]

**Comment by TonioGela** [comment]
We'll probably need a merge freeze for this, to avoid getting it merged before attributes migration

**Review by galales — APPROVED** [review]

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 5]
```diff
@@ -60,23 +59,22 @@ final case class ProcessApiServiceImpl(
 
     val result: Future[EService] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
-      clientSeed = Converter.convertToClientEServiceSeed(eServiceSeed, organizationId)
-      maybeEservice <- ReadModelQueries
-        .listEServices(
+      clientSeed = eServiceSeed.toDependency(organizationId)
+      maybeEservice <- ReadModelCatalogQueries
+        .getEServices(
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala:64]: shouldn't this go to CatalogManagementService?

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 6]
```diff
@@ -122,53 +121,50 @@ final case class ProcessApiServiceImpl(
...
+        states: Seq[CatalogDescriptorState],
+        agreementStates: Seq[PersistentAgreementState],
         offset: Int,
         limit: Int
       ): Future[PaginatedResult[CatalogItem]] = {
 
-        if (apiAgreementStates.isEmpty)
-          ReadModelQueries.listEServices(name, apiEServicesIds, apiProducersIds, apiStates, offset, limit)(readModel)
+        if (agreementStates.isEmpty)
+          ReadModelCatalogQueries.getEServices(name, eServicesIds, producersIds, states, offset, limit)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: same

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 7]
```diff
@@ -122,53 +121,50 @@ final case class ProcessApiServiceImpl(
...
             result                <-
               if (agreementEservicesIds.isEmpty)
-                Future.successful(ReadModelQueries.emptyResults[CatalogItem])
+                Future.successful(ReadModelCatalogQueries.emptyResults[CatalogItem])
               else
-                ReadModelQueries.listEServices(name, agreementEservicesIds, apiProducersIds, apiStates, offset, limit)(
-                  readModel
-                )
-
+                ReadModelCatalogQueries.getEServices(name, agreementEservicesIds, producersIds, states, offset, limit)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: same

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 8]
```diff
@@ -582,9 +603,9 @@ final case class ProcessApiServiceImpl(
     logger.info(operationLabel)
 
     val result: Future[EServiceConsumers] = for {
-      result <- ReadModelQueries.listConsumers(eServiceId, offset, limit)(readModel)
-      apiResults = result.results.map(_.toApi)
-    } yield EServiceConsumers(results = apiResults, totalCount = result.totalCount)
+      eServiceUuid <- eServiceId.toFutureUUID
+      result       <- ReadModelCatalogQueries.getConsumers(eServiceUuid, offset, limit)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: same

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**